docs/scenarios/springboot_2_to_3_security/README.md

```diff
@@ -0,0 +1,592 @@
+# Modernizing a Spring Boot 2 Application to Spring Boot 3 Using Konveyor AI
+
+## Goal
+
+This scenario demonstrates how **Konveyor AI (Kai)** can streamline the migration of a **Spring Boot 2** application to **Spring Boot 3**, **Java 17**, and **Jakarta EE**. We will use a combination of **default** and **custom rules** to:
+
+- Detect deprecated APIs like `WebSecurityConfigurerAdapter`
+- Update `javax.*` imports to `jakarta.*`
+- Ensure compliance with Java 17
+
+The sample application used is [`springboot_simplelogin`](https://github.com/savitharaghunathan/springboot_simplelogin), and the custom rules are located [here](https://github.com/savitharaghunathan/custom_rules/tree/main/springlogin).
+
+## Background
+
+Spring Boot 3 enforces significant changes including:
+
+- Jakarta EE 9+ migration (`javax.*` -> `jakarta.*`)
+- Java 17 as the baseline JDK
+- Removal of legacy security configurations
+
+These changes often break compilation and require manual refactoring. With Kai, developers can:
+
+- Automatically identify and resolve breaking changes
+- Get LLM-powered refactoring suggestions
+- Apply consistent migration patterns across projects using custom rules
+
+## Prerequisites
+
+- [VSCode](https://code.visualstudio.com/download)
+- [Git](https://git-scm.com/downloads)
+- [GenAI credentials](https://github.com/konveyor/kai/blob/main/docs/llm_selection.md#openai-service)
+- Java 17
+- Maven 3.9+
+- Sample App: [`springboot_simplelogin`](https://github.com/savitharaghunathan/springboot_simplelogin)
+- Kai VSCode IDE Extension `v0.1.0`
+
+## Step 1: Set Up Environment
+
+### 1.1 Clone the Sample App and custom rules
+
+```bash
+git clone https://github.com/savitharaghunathan/springboot_simplelogin.git
+
+git clone https://github.com/savitharaghunathan/custom_rules.git
+
+```
+
+### 1.2 Install Kai VSCode Extension
+
+Follow the [official guide](https://github.com/konveyor/kai/blob/main/docs/installation.md) for installation instructions.
+
+## Step 2: Understand the Custom Rules
+
+Before we run the analysis, let’s understand the custom rules we are going to use for migrating from Spring Boot 2 to Spring Boot 3, focusing specifically on the removal of `WebSecurityConfigurerAdapter` and discouraged configuration overrides.
+
+### 2.1. Custom Rule for Deprecated configure(AuthenticationManagerBuilder) Method
+
+The `SecurityConfig.java` class configures users and roles using the now-discouraged configure `(AuthenticationManagerBuilder auth)` method:
```

Comment:
small nit, to include 'configure' method name in the syntax highlight

change:
configure `(AuthenticationManagerBuilder auth)` method:

to:
`configure(AuthenticationManagerBuilder auth)` method: